### ISSUE TITLE
Klarna Gateway for WooCommerce

### ISSUE THREAD

**Issue by michaelbeil** [AffiliateWP/AffiliateWP#1125]
The Klarna gateway which includes Klarna Checkout and Klarna Invoice seems to have an issue making referrals in WooCommerce. The invoice option of the gateway works, but the checkout option does not. 

We are not 100% on this just yet, but this could be why AffiliateWP doesn't work with Klarna Checkout, considering that it works with, COD, Stripe, PayPal, etc.:

The Klarna gateway plugin for WooCommerce is missing the `woocommerce_checkout_order_processed` `do_action` hook that AffiliateWP and other plugins rely on. AffiliateWP adds a pending referral on this hook:

```php
add_action 'woocommerce_checkout_order_processed', array $this, 'add_pending_referral' , 10, 2 ;
```

We've scanned the plugin for the hook and all we found was a reference to the action hook in the readme file. It looks like it was added at some point, but now it doesn't exist anymore.  

```
2014.11.13
- version 1.9.0
* Tweak
- Added do_action'woocommerce_checkout_order_processed' after order has been created KCO, other plugins rely on this hook.  
```
We are attempting to get in touch with the Klarna Gateway developers. 

Let us :coffee:.  

Ref: https://secure.helpscout.net/conversation/171817344/27921/?folderId=205529

**Comment by michaelbeil** [comment]
**Update**: 

Download the plugin from WooThemes: https://www.woothemes.com/products/klarnafor free.  

In the settings for the plugin you can add this info to test the plugin with test mode activated:

***Eid:*** 200

***Shared secret:*** test

And for inserting payment info during testing you can use data from this link: https://developers.klarna.com/en/se+php/kpm/test-credentials.

**Comment by slobodan** [comment]
@michaelbeil not sure why this hook was removed, but I'm sorry about it.

Will send you the updated plugin in WooThemes support ticket right away.

**Comment by kevalmalde** [comment]
@slobodan Can I have access to that updated plugin?

**Comment by slobodan** [comment]
We pushed the update to WooThemes yesterday, it will most likely be available today. If it's not, please give me your email and I'll send it to you.

**Comment by michaelbeil** [comment]
@slobodan is this out in 2.1.3?

**Comment by slobodan** [comment]
@michaelbeil by out, do you mean removed?

The hook exists in 2.1.3 and this version is now available at WT.

**Comment by michaelbeil** [comment]
No, I was just verifying that the hook was in 2.1.3. Thanks @slobodan :smile:.

**Comment by michaelbeil** [comment]
This issue has been resolved in WooCommerce Gateway Klarna version 2.1.4.
